Lib/Protocols/IdHL7.pas

```diff
@@ -887,6 +1024,12 @@ procedure TIdHL7.StartServer;
     FServer.OnExecute := ServerExecute;
     FServer.OnDisconnect := ServerDisconnect;
     FServer.Active := True;
+    if FKeepAlive Then
+    Begin
+      d := $FFFFFFFF;
+      for i := 0 to FServer.Bindings.count - 1 Do
+        FServer.Bindings[i].SetSockOpt(Id_SOL_SOCKET, Id_SO_KEEPALIVE, iif(FKeepAlive, 1, 0));

```

Comment:
`TIdSocketHandle` has a public `SetKeepAliveValues()` method, you don't need to issue `SO_KEEPALIVE` directly here.

Reply:
Changed as requested
https://github.com/IndySockets/Indy/pull/442/commits/7bd8d29be4574608893391b97fcb794d959042b9